test/testMACILogic.js

```diff
@@ -223,111 +237,164 @@ describe('MACI', () => {
+        randomLeafPathIndexes
+      ] = stateTree.getPathUpdate(0)
+
+      // Get the vote options tree path elements
+      const [
+        user1VoteOptionsPathElements,
+        user1VoteOptionsPathIndexes
+      ] = user1VoteOptionTree.getPathUpdate(user1VoteOptionIndex)
+
+      const curVoteOptionTreeLeafRaw = user1VoteOptionTree.leavesRaw[user1VoteOptionIndex]
+
+      const stateTreeMaxIndex = BigInt(stateTree.nextIndex - 1)
+
+      const user1VoteOptionsTreeMaxIndex = BigInt(stateTree.nextIndex - 1)
+
+      const existingStateTreeLeaf = stateTree.leaves[user1StateTreeIndex]
+
+      const circuitInputs = stringifyBigInts({
+        'coordinator_public_key': coordinatorPk,
+        'message': user1Message,
+        'command': [
+          ...user1Command,
+          user1CommandSignature.R8[0],
+          user1CommandSignature.R8[1],
+          user1CommandSignature.S
+        ],
+        'msg_tree_root': msgTree.root,
+        'msg_tree_path_elements': msgTreePathElements,
+        'msg_tree_path_index': msgTreePathIndexes,
+        'vote_options_leaf_raw': curVoteOptionTreeLeafRaw,
+        'vote_options_tree_root': user1VoteOptionTree.root,
+        'vote_options_tree_path_elements': user1VoteOptionsPathElements,
+        'vote_options_tree_path_index': user1VoteOptionsPathIndexes,
+        'vote_options_max_leaf_index': user1VoteOptionsTreeMaxIndex,
+        'state_tree_leaf': existingStateTreeLeaf,
+        'state_tree_data': user1ExistingStateTreeData,
+        'state_tree_max_leaf_index': stateTreeMaxIndex,
+        'state_tree_root': stateTree.root,
+        'state_tree_path_elements': stateTreePathElements,
+        'state_tree_path_index': stateTreePathIndexes,
+        'random_leaf': randomPrivateKey(),
+        'random_leaf_path_elements': randomLeafPathElements,
+        'random_leaf_path_index': randomLeafPathIndexes,
+        'no_op': 1n,
+        'ecdh_private_key': babyJubJubPrivateKey(coordinatorSk),
+        'ecdh_public_key': ephemeralPk
+      })
 
-      const circuitInput = {
-        cmd_tree_root: stringifyBigInts(cmdTree.root),
-        cmd_tree_path_elements: stringifyBigInts(cmdTreePathElements),
-        cmd_tree_path_index: stringifyBigInts(cmdTreePathIndex),
-        state_tree_root: stringifyBigInts(stateTree.root),
-        state_tree_path_elements: stringifyBigInts(stateTreePathElements),
-        state_tree_path_index: stringifyBigInts(stateTreePathIndex),
-        encrypted_data: stringifyBigInts(user2NewEncryptedMsg),
-        existing_public_key: stringifyBigInts(user2PublicKey),
-        existing_state_tree_leaf: stringifyBigInts(user2Leaf),
-        ecdh_private_key: stringifyBigInts(ecdhPrivateKey)
-      }
+      const wasmBn128 = await buildBn128()
+      const zkSnark = groth
 
-      const witness = circuit.calculateWitness(circuitInput)
+      const circuit = new Circuit(updateStateTreeCircuitDef)
+      const witness = circuit.calculateWitness(circuitInputs)
```

Comment:
I think we should also get the output (the new state root) and test whether it matches our off-chain Merkle root.